packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx

```diff
@@ -162,6 +188,36 @@ const ProjectManagementPanel: FC = () => {
                     projectUuid={deletingProjectUuid}
                 />
             ) : null}
+            {isCreatePreviewOpen && lastProject && (
+                <Modal
+                    opened={isCreatePreviewOpen}
+                    onClose={() => setIsCreatePreview(false)}
+                    title={`Create preview from ${lastProject.name}`}
+                >
+                    <Text>
+                        This will create a preview project from $
+                        {lastProject.name}. The new project will have the same
+                        connections and credentials.
+                        <Group mt="sm" mb="sm">
+                            {' '}
+                            Preview name
+                            {/* TODO use inputWrapper*/}
+                            <Input value={`Preview of ${lastProject.name}`} />
+                        </Group>
+                        <Button
+                            onClick={() => {
```

Comment:
I would add a loading state here while the previewProject is being created, and probably make the `onClick` async with the `mutateAsync`